### PR TITLE
CI: give dispatch-rebottle comment access

### PR DESCRIPTION
**Pull request by SMillerDev**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
Currently it can't comment on failures

### CODE REVIEW

**Comment by carlocab** [comment]
Should probably put that here:

https://github.com/Homebrew/homebrew-core/blob/c01caa6aed75be742b3a2b1ff2bb2bba19b90329/.github/workflows/dispatch-rebottle.yml#L58-L60

Not sure why it's failing though, since we already have `issues: write`

File: .github/workflows/dispatch-rebottle.yml [hunk 1]
```diff
@@ -28,6 +28,7 @@ on:
 
 permissions:
   contents: read
+  pull-requests: write
```
Comment (ZhongRuoyu) [inline .github/workflows/dispatch-rebottle.yml:31]: May want to add this to the `bottle` job's `permissions` instead. Some permissions are already set there.

https://github.com/Homebrew/homebrew-core/blob/c01caa6aed75be742b3a2b1ff2bb2bba19b90329/.github/workflows/dispatch-rebottle.yml#L57-L60

By the way, the `bottle` job in `dispatch-build-bottle` workflow may need this as well.

**Comment by ZhongRuoyu** [comment]
> Not sure why it's failing though, since we already have `issues: write`

I *think* posting comments under PRs may require additional permissions. Although the issues API was used (https://github.com/Homebrew/actions/blob/master/post-comment/main.js).

**Comment by carlocab** [comment]
> > Not sure why it's failing though, since we already have `issues: write`
> 
> I _think_ posting comments under PRs may require additional permissions. Although the issues API was used (https://github.com/Homebrew/actions/blob/master/post-comment/main.js).

It should be the [same endpoint](https://docs.github.com/en/rest/issues/comments?apiVersion=2022-11-28#create-an-issue-comment) though 🤷

**Comment by SMillerDev** [comment]
Maybe it needs PR read? The error isn't very clear on the problem:

> Error: Resource not accessible by integration

**Pull request closed without merging**